From 571b7270710f642fe5bd2a9d18e373f22c1666b1 Mon Sep 17 00:00:00 2001
From: chenyanxzhu <chenyanx.zhu@intel.com>
Date: Wed, 26 Oct 2022 09:53:47 +0800
Subject: [PATCH] Add more sepolicy rules for mediatranscoding

Fix avc denied error for mediatranscoding, when running cts
case CtsMediaProviderTranscodeTests. 'mediatranscoding' is a
system defined private domain.

As media.transcoding service need access to gpu device and find
surfaceflinger, added sepolicies to fix the CTS issue.

Tracked-On: OAM-104121
Signed-off-by: chenyanxzhu <chenyanx.zhu@intel.com>
---
 system_ext/private/mediatranscoding.te | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 system_ext/private/mediatranscoding.te

diff --git a/system_ext/private/mediatranscoding.te b/system_ext/private/mediatranscoding.te
new file mode 100644
index 0000000..82d29f5
--- /dev/null
+++ b/system_ext/private/mediatranscoding.te
@@ -0,0 +1,6 @@
+# mediatranscoding - daemon for transcoding video and image.
+allow mediatranscoding surfaceflinger_service:service_manager find;
+allow mediatranscoding gpu_device:dir { open read search };
+allow mediatranscoding gpu_device:chr_file { getattr ioctl map open read write };
+allow mediatranscoding graphics_device:chr_file { getattr };
+
-- 
2.38.1

